api/src/s2-api.ts

```diff
@@ -90,7 +90,62 @@ async function getPaper(s2Id: string, apiKey?: string): Promise<Paper | undefine
   );
   const response = await axios.get<S2ApiPaper>(`${apiOrigin}/paper/${s2Id}`, conf);
   const paper = toPaper(s2Id, response.data);
-  cache.set(s2Id, paper);
+  // only cache if the request is successful to avoid papers being permanent errors
+  if (response.status === 200) {
```

Comment:
Wouldn't a non-200 throw in the `await` above?

Reply:
Come to think of it, I think this issue was actually addressed by using the LRU cache in here (it used to be that this was just an in-memory map that would store the response for each paper for all eternity). Going to revert this change.